shinydashboard/lantern/modules/smartresponsemodule.R

```diff
@@ -48,6 +48,85 @@ smartresponsemodule <- function(
     res
   })
 
+  selected_smart_count_total <- reactive ({
+    all <- endpoint_export_tbl
+    req(sel_fhir_version(), sel_vendor())
+    if (sel_fhir_version() != ui_special_values$ALL_FHIR_VERSIONS) {
+      all <- all %>% filter(fhir_version == sel_fhir_version())
+    }
+    if (sel_vendor() != ui_special_values$ALL_DEVELOPERS) {
+      all <- all %>% filter(vendor_name == sel_vendor())
+    }
```

Comment:
It seems like you could possibly make this into a function that all of these other functions call so you don't have to write it out every time? something like:
```
get_filtered_data <- function(table_val) {
   all <- table_val
   req(sel_fhir_version(), sel_vendor())
   ....
   all
}
```
at least I think that would be possible in R?